### PR TITLE
History & Homepage 

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Features added 
- Adding video to history once the user visits a single video page.
- Added home page & category section.

### CODE REVIEW

**Review by vijaytembugade — APPROVED** [review]
Good UI. Code looks clean .
Good work.

File: src/pages/single-video/SingleVideo.jsx [hunk 1]
```diff
@@ -32,10 +32,16 @@ const SingleVideo = () => {
...
+  const {
+    historyState: { history },
+    addToHistory,
+  } = useWatchHistory();
+
   const navigate = useNavigate();
   const video = videos.find(eachVideo => eachVideo._id === videoId);
   const isLiked = likedList.find(eachVideo => eachVideo._id === videoId);
   const isPresentInWatchLater = watchLaterVideos.find(eachVideo => eachVideo._id === videoId);
+  const isPresentInHistory = history.find(eachVideo => eachVideo?._id === videoId);
```
Comment (poornima-alugubelly) [inline src/pages/single-video/SingleVideo.jsx]: u can abstract out line 43 and 44 by making a function which checks whether a video is in a playlist

**Review by poornima-alugubelly — APPROVED** [review]
Good clean code 👍

File: src/pages/single-video/SingleVideo.jsx (hunk 1 above)
Comment (msdeshmukh009) [reply src/pages/single-video/SingleVideo.jsx]: Done. Thanks for the review.

**Pull request merged**